src/cachemap.h

```diff
@@ -118,7 +118,7 @@ class CacheMap
             PruneLast();
         }
         listItems.push_front(item_t(key, value));
-        mapIndex[key] = listItems.begin();
+        mapIndex.insert(std::make_pair(key, listItems.begin()));
```

Comment:
Why not `emplace(key, listItems.begin())` instead of insert? Would remove one unnecessary copy of the objects.

Reply:
Good point!